node/test/trace/basic_server.js

```diff
@@ -70,17 +70,27 @@ test('basic tracing test', function (assert) {
     server.handler.register('/top_level_endpoint', function (req, res) {
         logger.info("top level sending to subservice");
         setTimeout(function () {
-            server
-                .request({
-                    host: '127.0.0.1:4042', 
-                    serviceName: 'subservice', 
-                    parentSpan: req.span,
-                    trace: true
-                }).send('/foobar', 'arg1', 'arg2', function (err, subRes) {
-                    logger.info("top level recv from subservice");
+            var servReq = server.request({
+                host: '127.0.0.1:4042',
+                serviceName: 'subservice',
+                parentSpan: req.span,
+                trace: true});
+            var peers = server.peers.values();
+            var ready = new CountedReadySignal(peers.length);
+            peers.forEach(function each(peer) {
+                if (peer.isConnected()) {
+                    ready.signal();
+                } else {
+                    peer.connect().on('identified', ready.signal);
+                }
```

Comment:
And here’s another situation where it looks like we should use signals instead of events. `peer.connectedSignal.on(…)`, to simplify monitoring the connected state, whether it happened before or will happen eventually, whether it needs to be initiated (peer.connect()) or whether it has already been initiated (peer.isConnected()), and perhaps even capture and replay errors to new observers introduced while connecting via the `peer.connect(cb(err))` interface. cc @jcorbin @Raynos